sopel/modules/reddit.py

```diff
@@ -27,10 +27,13 @@
 
 domain = r'https?://(?:www\.|old\.|pay\.|ssl\.|[a-z]{2}\.)?reddit\.com'
 subreddit_url = r'%s/r/([\w-]+)/?$' % domain
-post_url = r'%s/r/\S+?/comments/([\w-]+)(?:/[\w%%]+)?/?' % domain
+post_or_comment_url = (
+    domain +
+    r'/r/\S+?/comments/(?P<submission>[\w-]+)'
+    r'(?:/?(?:[\w%%]+/(?P<comment>[\w-]+))?)'
```

Comment:
@Duckle29 pointed out on IRC that this probably isn't necessary if no `%` interpolation is happening.

```suggestion
    r'(?:/?(?:[\w%]+/(?P<comment>[\w-]+))?)'
```